### PR TITLE
fix: optimize sub table creation

### LINKED ISSUE #152: Timed out when creating a partitioned table with a large number of partitions
**Describe this problem**
When create partition table with a large number of partitions, the client's request will time out. 

**Steps to reproduce**
Execute create partitioned table like this:
```
CREATE TABLE `demo` (`name` string TAG, `value` double NOT NULL, `t` timestamp NOT NULL, TIMESTAMP KEY(t))
    PARTITION BY KEY(name) PARTITIONS 512
    ENGINE=Analytic with (enable_ttl='false')
```

**Expected behavior**
Successfully completed the creation of the partition table in a short time.

**Additional Information**
In the current implementation, the sub-table creation of the partition table is serial, maybe we need to optimize the sub-table creation process.

### PR DESCRIPTION
**Pull request by baojinri**
# Which issue does this PR close?

Closes #152 

# Rationale for this change
In order to avoid timeout when creating a partitioned table with a large number of partitions
 

# What changes are included in this PR?
-Optimize the sub table creation process

# Are there any user-facing changes?
Someone create a partitioned table with a large number of partitions

# How does this change test
unit test & integration test

### CODE REVIEW

File: server/coordinator/procedure/ddl/createpartitiontable/create_partition_table.go [hunk 1]
```diff
@@ -219,35 +219,62 @@ func createDataTablesCallback(event *fsm.Event) {
...
 			PartitionInfo: storage.PartitionInfo{},
-		})
+		}
+		shardTableMetaDatas[subTableShard.ShardInfo.ID] = append(shardTableMetaDatas[subTableShard.ShardInfo.ID], tableMetaData)
+	}
+	shardVersionsCh := make(map[storage.ShardID]chan uint64, len(shardTableMetaDatas))
+	for shardID, tableMetaDatas := range shardTableMetaDatas {
+		shardVersion := shardVersions[shardID]
+		shardVersionsCh[shardID] = make(chan uint64)
+		go createDataTables(event, shardID, tableMetaDatas, shardVersion, shardVersionsCh[shardID])
```
Comment (ZuLiangWang) [inline server/coordinator/procedure/ddl/createpartitiontable/create_partition_table.go:235]: What will happen if invoke `createDataTables` failed? It seems that we could not get error and stop creating table. Maybe we need use `errorgroup` to get `createDataTables` result.

File: server/coordinator/procedure/ddl/createpartitiontable/create_partition_table.go [hunk 2]
```diff
@@ -219,35 +219,62 @@ func createDataTablesCallback(event *fsm.Event) {
...
-			ShardID:     subTableShard.ShardInfo.ID,
-			CurrVersion: req.p.relatedVersionInfo.ShardWithVersion[subTableShard.ShardInfo.ID] + 1,
-			PrevVersion: req.p.relatedVersionInfo.ShardWithVersion[subTableShard.ShardInfo.ID],
+			ShardID:     shardID,
+			CurrVersion: shardVersion + 1,
+			PrevVersion: shardVersion,
 		}
 
-		if err := ddl.CreateTableOnShard(req.ctx, params.ClusterMetadata, params.Dispatch, subTableShard.ShardInfo.ID, ddl.BuildCreateTableRequest(result.Table, shardVersionUpdate, params.SourceReq)); err != nil {
+		if err := ddl.CreateTableOnShard(req.ctx, params.ClusterMetadata, params.Dispatch, shardID, ddl.BuildCreateTableRequest(result.Table, shardVersionUpdate, params.SourceReq)); err != nil {
```
Comment (ZuLiangWang) [inline server/coordinator/procedure/ddl/createpartitiontable/create_partition_table.go:267]: We need to support the user to rebuild the table when the table creation fails. You can refer to this pr https://github.com/CeresDB/ceresmeta/pull/179, which supports repeated table creation for ordinary tables.

Comment (baojinri) [reply server/coordinator/procedure/ddl/createpartitiontable/create_partition_table.go:269]: I will solve this problem in the future pr

Comment (ZuLiangWang) [reply server/coordinator/procedure/ddl/createpartitiontable/create_partition_table.go:269]: Add a todo comment here.

**Review by ZuLiangWang — APPROVED** [review]
LGTM

**Pull request merged**